### PR TITLE
Implement `x[y]` record indexing

### PR DESCRIPTION
**Pull request by raxyte**
Implements indexing records with expressions.

### CODE REVIEW

**Review by IyeOnline — APPROVED** [review]

File: changelog/next/features/4795--record-indexing.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Indexing records with string expressions is now supported.
```
Comment (mavam) [inline changelog/next/features/4795--record-indexing.md:1]: Can you provide a brief example in the changelog?

File: tenzir/integration/tests/expression.bats [hunk 2]
```diff
@@ -55,6 +55,17 @@ z = [...x, 3, ...y, ...42]
...
 
+@test "record indexing" {
+  echo '
+  {"x": {"x": 1}, "y": "x"}
+  {"x": {"x": "1"}, "y": "x"}
+  {"x": {"x": 1}}
+  {"x": {}, "y": "x"}
+  {"y": "x"}
+  {}' |
+    check tenzir 'read_json | z = x[y]'
```
Comment (mavam) [inline tenzir/integration/tests/expression.bats:66]: Have you considered this?

```
from [
  {...},
  {...},
  {...},
  {...},
]
z = x[y]
```

**Comment by mavam** [comment]
I just saw automerge is already one. My comments are mostly nits; feel free to ignore.

File: tenzir/integration/tests/expression.bats (hunk 2 above)
Comment (raxyte) [reply tenzir/integration/tests/expression.bats:66]: You'll get type clashes in `from` that way.

Comment (mavam) [reply tenzir/integration/tests/expression.bats:66]: Ah yes. I actually ran into this as well!

I had issues with:
```
from [
  {x: 1, y: "foo"},
  {x: 2, y: "bar"},
  {x: 1},
  {x: 2},
  {y: "foo"},
]
taste 1
```

Comment (raxyte) [reply tenzir/integration/tests/expression.bats:66]: That should work fine, types of `x` and `y` don't change in there. Maybe you had a slightly different example.

File: changelog/next/features/4795--record-indexing.md (hunk 1 above)
Comment (raxyte) [reply changelog/next/features/4795--record-indexing.md:1]: We couldn't come up with a concise example that showed the expression evaluating to string and then accessing the record.

**Pull request merged**